### PR TITLE
gnrc_ipv6_ext_frag: check return value of msg_try_send()

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Addresses parts of #15521. Not checking the return value of `msg_try_send()` could indeed lead to a leak in the packet buffer.

### Testing procedure
I tried to provide tests for the corner case, but RIOT makes it quite hard to mock filled message queues :-/. However, `tests/gnrc_ipv6_ext_frag` should still pass never the less **NOTE THE SUDO! This test is not run on Murdock**:

```sh
sudo ./dist/tools/tapsetup/tapsetup   # if you haven't already opened a TAP interface
make -C tests/gnrc_ipv6_ext_frag -j
sudo make -C tests/gnrc_ipv6_ext_frag test
```

Of course this also should work with any board supporting `ethos`.

### Issues/PRs references
Addresses parts of #15521

### CODE REVIEW

File: sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c [hunk 1]
```diff
@@ -255,7 +255,12 @@ void gnrc_ipv6_ext_frag_send(gnrc_ipv6_ext_frag_send_t *snd_buf)
     /* tell gnrc_ipv6 to send the above prepared fragment */
     msg.type = GNRC_IPV6_EXT_FRAG_SEND;
     msg.content.ptr = to_send;
-    msg_try_send(&msg, gnrc_ipv6_pid);
+    if (msg_try_send(&msg, gnrc_ipv6_pid) <= 0) {
+        DEBUG("ipv6_ext_frag: Unable to send fragment, canceling fragmentation\n");
+        gnrc_pktbuf_release(to_send);
+        _snd_buf_free(snd_buf);
+        return;
```
Comment (benpicco) [inline sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c:262]: can we propagate that error up the stack?

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c:262]: Of course!

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c:263]: Done! Made it a separate commit, so the other cases of `_snd_buf_free()` are included as well.

File: sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c [hunk 2]
```diff
@@ -291,13 +300,13 @@ static void _snd_buf_del(gnrc_ipv6_ext_frag_send_t *snd_buf)
...
 
-static void _snd_buf_free(gnrc_ipv6_ext_frag_send_t *snd_buf)
+static void _snd_buf_free(gnrc_ipv6_ext_frag_send_t *snd_buf, int errno)
 {
     if (snd_buf->per_frag) {
         gnrc_pktbuf_release(snd_buf->per_frag);
     }
     if (snd_buf->pkt) {
-        gnrc_pktbuf_release(snd_buf->pkt);
+        gnrc_pktbuf_release(snd_buf->pkt, errno);
```
Comment (benpicco) [inline sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c:309]:
```suggestion
        gnrc_pktbuf_release_error(snd_buf->pkt, errno);
```

File: sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c [hunk 3]
```diff
@@ -291,13 +300,13 @@ static void _snd_buf_del(gnrc_ipv6_ext_frag_send_t *snd_buf)
     snd_buf->pkt = NULL;
 }
 
-static void _snd_buf_free(gnrc_ipv6_ext_frag_send_t *snd_buf)
+static void _snd_buf_free(gnrc_ipv6_ext_frag_send_t *snd_buf, int errno)
 {
     if (snd_buf->per_frag) {
         gnrc_pktbuf_release(snd_buf->per_frag);
```
Comment (benpicco) [inline sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c:306]: Should we also use `gnrc_pktbuf_release_error()` here?

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c:306]: `per_frag` is derived from `pkt`, so I don't think that is necessary. A subscriber would then only get duplicate messages:

https://github.com/RIOT-OS/RIOT/blob/3234b918e361d273b9cfef3c123653b174f0abc0/sys/net/gnrc/network_layer/ipv6/ext/frag/gnrc_ipv6_ext_frag.c#L112-L127

**Comment by miri64** [comment]
May I squash?

**Comment by benpicco** [comment]
Murdock is not yet happy

**Comment by miri64** [comment]
It was your idea to make him unhappy :P

**Comment by miri64** [comment]
Fixed!

**Comment by benpicco** [comment]
And the tests works too - please squash.

**Review by benpicco — APPROVED** [review]

**Comment by miri64** [comment]
Squashed! :-)

**Pull request merged**